src/clientSideScene/CameraControls.ts

```diff
@@ -773,6 +773,80 @@ export class CameraControls {
     })
   }
 
+  async updateCameraToAxis(
+    axis: 'x' | 'y' | 'z' | '-x' | '-y' | '-z' | 'reset'
+  ): Promise<void> {
+    if (axis === 'reset') {
+      await this.resetCameraPosition()
+      return
+    }
+
+    const distance = this.camera.position.distanceTo(this.target)
+
+    let vantage = { x: 0, y: 0, z: 0 }
+    let up = { x: 0, y: 0, z: 0 }
+
+    if (axis === 'x') {
+      vantage = { x: distance, y: 0, z: 0 }
+      up = { x: 0, y: 0, z: 1 }
+    } else if (axis === 'y') {
+      vantage = { x: 0, y: distance, z: 0 }
+      up = { x: 0, y: 0, z: 1 }
+    } else if (axis === 'z') {
+      vantage = { x: 0, y: 0, z: distance }
+      up = { x: -1, y: 0, z: 0 }
+    } else if (axis === '-x') {
+      vantage = { x: -distance, y: 0, z: 0 }
+      up = { x: 0, y: 0, z: 1 }
+    } else if (axis === '-y') {
+      vantage = { x: 0, y: -distance, z: 0 }
+      up = { x: 0, y: 0, z: 1 }
+    } else if (axis === '-z') {
+      vantage = { x: 0, y: 0, z: -distance }
+      up = { x: -1, y: 0, z: 0 }
+    }
+
+    await this.engineCommandManager.sendSceneCommand({
+      type: 'modeling_cmd_req',
+      cmd_id: uuidv4(),
+      cmd: {
+        type: 'default_camera_look_at',
+        center: { x: 0, y: 0, z: 0 },
+        vantage: vantage,
+        up: up,
+      },
+    })
+    await this.engineCommandManager.sendSceneCommand({
+      type: 'modeling_cmd_req',
+      cmd_id: uuidv4(),
+      cmd: {
+        type: 'default_camera_get_settings',
+      },
+    })
+  }
+
+  async resetCameraPosition(): Promise<void> {
+    await this.engineCommandManager.sendSceneCommand({
+      type: 'modeling_cmd_req',
+      cmd_id: uuidv4(),
+      cmd: {
+        type: 'default_camera_look_at',
+        center: { x: 0, y: 0, z: 0 },
+        vantage: { x: 0, y: -128, z: 64 },
+        up: { x: 0, y: 0, z: 1 },
+      },
+    })
+    await this.engineCommandManager.sendSceneCommand({
+      type: 'modeling_cmd_req',
+      cmd_id: uuidv4(),
+      cmd: {
+        type: 'zoom_to_fit',
+        object_ids: [], // leave empty to zoom to all objects
+        padding: 0.2, // padding around the objects
+      },
+    })
+  }
```

Comment:
Sorry I couldn't help but make a commit, just cause I wanted to see this working.

All I did is change the vantage to where the camera is on boot up `vantage: { x: 0, y: -128, z: 64 },` The most important thing here is the ratio of y and z (double y).

But immediately followed up by a zoom to fit means it will look exactly the same as when you refresh the page.

Note: `default_camera_get_settings` is not needed because we already subscribe to the camera payload of `zoom_to_fit`

Reply:
Okay this sounds like a good work around, but maybe the zoom to fit should be robust against this kind of thing?

Maybe I'll mess with it and raise something with the engine team because I'm not sure it should require us to finagle like this.